src/Microsoft.TestPlatform.PlatformAbstractions/net451/System/ProcessStartInfoExtensions.cs

```diff
@@ -15,7 +15,7 @@ public static class ProcessStartInfoExtensions
         /// <param name="value">Environment Variable value.</param>
         public static void AddEnvironmentVariable(this ProcessStartInfo startInfo, string name, string value)
         {
-            startInfo.EnvironmentVariables.Add(name, value);
+            startInfo.EnvironmentVariables[name] = value;
```

Comment:
Have we tested that this doesn't throw exception on overriding?
Earlier even add was not throwing exception.